File: SNetUnity/SNetClient.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System.Net;
using System;
public enum ClientStatus
{
    UNKNOWN=1,
    CONNECTED=2,
    DISCONNECTED=3
}
public class SNetClient
{
    public IPEndPoint endPoint;
    byte[] recvbuff = new byte[1024];
    int buffSize=0;
    public event Action<SPacket> OnPacketRecived;
    //ClientStatus status = ClientStatus.UNKNOWN;
    //DateTime lastSeen;
    public SNetClient(IPEndPoint ep)
    {
        this.endPoint = ep;
        //lastSeen = DateTime.Now;
        //lastSeen=lastSeen.AddSeconds(-10);
    }
    public void ReciveBytes(byte[] bytes)
    {
        Array.Copy(bytes, 0, recvbuff, buffSize, bytes.Length);
        buffSize += bytes.Length;
        int toRead = SPacket.ToRead(recvbuff, 0, buffSize);
        Debug.Log(toRead);
        if (toRead<=buffSize)
        {
            SPacket packet = new SPacket(recvbuff);
            Debug.Log(packet.id);
            PacketReceived(packet);
            Array.Copy(recvbuff, toRead, recvbuff, 0, recvbuff.Length-toRead);
            buffSize -= toRead;
        }
    }
    void PacketReceived(SPacket packet)
    {
        //if (packet.id == SPacketId.STATUS)
        //{
        //    ClientStatus s =(ClientStatus) packet.ReadInt32();
        //    status = s;
        //    lastSeen = DateTime.Now;
        //    Debug.Log(s);
        //}
        if (OnPacketRecived != null)
        {
            TaskManager.instance.AddTask(()=>OnPacketRecived(packet));
        }
    }
    //public void CheckStatus()
    //{
    //    if (lastSeen.AddSeconds(5) < DateTime.Now)
    //    {
    //        status = ClientStatus.UNKNOWN;
    //    }
    //}
}

File: SNetPython/snetclient.py
import spacket as sp
import socket
import struct
from enum import Enum
class ClientStatus(Enum):
    UNKNOWN=0
    CONNECTED=1
    DISCONNECTED=2
class SNetClient:
    def __init__(self, endpoint):
        self.endpoint = endpoint
        self.recv_buff = bytearray(1024)
        self.buff_size = 0
        # self.client_status=ClientStatus.UNKNOWN
        self.on_packet_received = None

    def receive_bytes(self, bytes_data):
        self.recv_buff[self.buff_size:self.buff_size + len(bytes_data)] = bytes_data
        self.buff_size += len(bytes_data)
        to_read = sp.SPacket.to_read(self.recv_buff, 0, self.buff_size)

        print(to_read)  # Print for testing

        if to_read <= self.buff_size:
            packet = sp.SPacket(-1,self.recv_buff)
            print(packet.id)  # Print for testing
            self.packet_received(packet)
            self.recv_buff[:len(self.recv_buff) - to_read] = self.recv_buff[to_read:]
            self.buff_size -= to_read

    def packet_received(self, packet):
        # if(packet.id==sp.SPacketId.STATUS):
            # self.status=ClientStatus(packet.read_int32())
            # print(self.status)
        if self.on_packet_received is not None:
            self.on_packet_received(packet)